sources/common/inc/LogParser.h

```diff
@@ -49,9 +51,15 @@ namespace PARSER
     {
         public:
 
-            /// Returns reference to the signal database.
+            /// Returns the const reference to the signal database.
             const SIGNAL::SignalDb &GetSignalDb() const
             {
+                return const_cast<SIGNAL::SignalDb &>((this)->GetSignalDb());
```

Comment:
What is this abomination?
What is it needed for?
How is it now needed for something if it wasn't before?